src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.RegionBuilder.cs

```diff
@@ -40,15 +37,30 @@ public RegionBuilder(ControlFlowRegionKind kind, ITypeSymbol exceptionType = nul
                 IsStackSpillRegion = isStackSpillRegion;
             }
 
-            public bool IsEmpty => FirstBlock == null;
+            [MemberNotNullWhen(false, nameof(FirstBlock), nameof(LastBlock))]
+            public bool IsEmpty
+            {
+                get
+                {
+                    Debug.Assert((FirstBlock == null) == (LastBlock == null));
+#pragma warning disable CS8775 // LastBlock is null when exiting, verified equivalent to FirstBlock above
+                    return FirstBlock == null;
```

Comment:
I'm trying to understand this a bit: the compiler is issuing a warning because we're not explicitly nulling LastBlock and so it doesn't know it's state?

Reply:
Correct.